### PR TITLE
Added capability to display labels for each thumbnail

### PR DESCRIPTION
**Pull request by ganjash**
Hi xiaolin,
I added a capability that gives small labels below each thumbnail, it's taken as property and is optional to display. Please mention any opinions or do code changes on top of it.

### CODE REVIEW

File: src/ImageGallery.jsx [hunk 1]
```diff
@@ -582,11 +582,15 @@ export default class ImageGallery extends React.Component {
 
             onTouchStart={event => this.slideToIndex.call(this, index, event)}
             onClick={event => this.slideToIndex.call(this, index, event)}>
-
-            <img
-              src={item.thumbnail}
-              alt={item.thumbnailAlt}
-              onError={onThumbnailError.bind(this)}/>
+            <div style={{ width: '63px' }}>
```
Comment (xiaolin) [inline src/ImageGallery.jsx]: you shouldn't need another `<div>` to wrap the `` its already wrapped by `<a>`

File: src/ImageGallery.jsx [hunk 2]
```diff
@@ -582,11 +582,15 @@ export default class ImageGallery extends React.Component {
...
+              <img
+                src={item.thumbnail}
+                alt={item.thumbnailAlt}
+                onError={onThumbnailError.bind(this)}/>
+               <span style={{ fontSize: '0.8em', color: 'black' }}>
```
Comment (xiaolin) [inline src/ImageGallery.jsx]: Define a className and put styles in `scss` file. Only use inline when creating dynamic styles.

File: src/ImageGallery.jsx [hunk 3]
```diff
@@ -582,11 +582,15 @@ export default class ImageGallery extends React.Component {
...
+                 {item.label}
```
Comment (xiaolin) [inline src/ImageGallery.jsx]: This should be consistent with the other thumbnail properties being called `thumbnailLabel`

**Comment by xiaolin** [comment]
@ganjash Thanks for the PR, I left some notes. Please drop a screen shot of the thumbnail label so I can see how you styled it.

Thanks.

File: src/ImageGallery.jsx (hunk 2 above)
Comment (ganjash) [inline src/ImageGallery.jsx]: I tried adding styles with className in scss xiaolin, But it didn't work. The build was successful and i could see my style classNames in build/image_gallery.css The stylings i added were not displayed in the browser.
code snippet:

src/image-gallery.scss
`.image-gallery-div-thumbnail {
  width: 63px;
}

.image-gallery-label {
  font-size: 0.8em;
  color: $ig-black;
}
`
src/ImageGallery.jsx
`          <a
            onMouseOver={this._handleMouseOverThumbnails.bind(this, index)}
            onMouseLeave={this._handleMouseLeaveThumbnails.bind(this, index)}
            key={index}
            className={
              'image-gallery-thumbnail' +
              (currentIndex === index ? ' active' : '') +
              thumbnailClass
            }

            onTouchStart={event => this.slideToIndex.call(this, index, event)}
            onClick={event => this.slideToIndex.call(this, index, event)}>
            <div className="image-gallery-div-thumbnail">
            
            <span className="image-gallery-label">
              {item.label}
            </span>
            </div>
          </a>
`

Comment (ganjash) [inline src/ImageGallery.jsx]: `.image-gallery-div-thumbnail {
  width: 63px;
}

.image-gallery-label {
  font-size: 0.8em;
  color: $ig-black;
}
`

Comment (xiaolin) [inline src/ImageGallery.jsx]: Thats weird, double check your build, it should work.

File: src/ImageGallery.jsx (hunk 1 above)
Comment (xiaolin) [inline src/ImageGallery.jsx]: I forgot to mention, don't use arbitrary width like `63px` the label should be responsive with the parent container.

**Comment by ganjash** [comment]
screenshot of how it looks:

screen shot 2016-08-05 at 2 36 53 pm

**Comment by xiaolin** [comment]
@ganjash after seeing the screenshot I'm curious about your use case. 

Why not use `description`  to give a descriptive meaning to the picture? 

Thumbnails are generally small and does not fit text well. If you have a longer text string say `Plane that is blue` then this wouldn't work well.

Is there a reason you want to label the thumbnails?

**Comment by ganjash** [comment]
@xiaolin , in my use case, i am supposed to show dates for images: MM/DD. So, you can see which images is taken when at one glance in thumbnails and directly click on the images. Since, date would restrict the purpose of label, I'm passing "mm/dd" as just a string to label each thumbnail.

I though labeling would be helpful to give a **nick name, or small tag**, not long descriptions.
Pic of my use case wud be like this:

screen shot 2016-08-05 at 5 10 41 pm

**Comment by xiaolin** [comment]
I see, that makes sense.

**Comment by xiaolin** [comment]
If you update the PR I can merge your feature.

**Comment by ganjash** [comment]
sure will do it @xiaolin

**Comment by ganjash** [comment]
I made it responsive @xiaolin . You may merge now.

File: src/ImageGallery.jsx [hunk 4]
```diff
@@ -582,11 +582,13 @@ export default class ImageGallery extends React.Component {
...
-
-            <img
-              src={item.thumbnail}
-              alt={item.thumbnailAlt}
-              onError={onThumbnailError.bind(this)}/>
+              <img
+                src={item.thumbnail}
+                alt={item.thumbnailAlt}
+                onError={onThumbnailError.bind(this)}/>
+              <div >
```
Comment (xiaolin) [inline src/ImageGallery.jsx]: Add a class name called `image-gallery-thumbnail-label` so we can style this specifically

File: src/image-gallery.scss [hunk 5]
```diff
@@ -152,6 +152,15 @@ $ig-background-black: rgba(0, 0, 0, .4) !default;
   display: inline-block;
   padding-right: 5px;
 
+  div {
```
Comment (xiaolin) [inline src/image-gallery.scss]: We don't want to style all of our divs to have this style. Use `image-gallery-thumbnail-label`

File: src/ImageGallery.jsx (hunk 4 above)
Comment (ganjash) [inline src/ImageGallery.jsx]: do you mean style div tag alone specifically ?

Comment (xiaolin) [inline src/ImageGallery.jsx]: Yes, so if you add a `className` we can target the style only to that div

File: src/image-gallery.scss (hunk 5 above)
Comment (ganjash) [inline src/image-gallery.scss]: added the styling and classname @xiaolin . Please check

**Comment by ganjash** [comment]
made the suggested stylings for thumbails

**Pull request merged**